cpp/independent_set/IndependentSet.cpp

```diff
@@ -0,0 +1,318 @@
+/*
+ * IndependentSet.cpp
+ *
+ *  Created on: 30.12.2022
+ *      Author: Artur Salawa
+ */
+
+#include <independent_set/IndependentSet.hpp>
+
+#include <networkit/auxiliary/BucketPQ.hpp>
+#include <networkit/graph/GraphTools.hpp>
+
+#include <limits>
+#include <bitset>
+
+namespace Koala {
+
+void printGraph(const NetworKit::Graph* graph) { // TODO: its just debug
+    graph->forNodes([&](NetworKit::node v) {
+        std::cout << v << " ";
+    });
+    std::cout << std::endl;
+    graph->forEdges([&](NetworKit::node u, NetworKit::node v) {
+        std::cout << "{" << u << "," << v << "}, ";
+    });
+    std::cout << std::endl;
+}
+
+std::vector<NetworKit::node> IndependentSet::getNeighbors(NetworKit::node v) const {
+    std::vector<NetworKit::node> neighbors;
+    graph->forNeighborsOf(v, [&](NetworKit::node u) {
+        neighbors.push_back(u);
+    });
+    return neighbors;
+}
+
+std::vector<NetworKit::node> IndependentSet::getNeighborsPlus(NetworKit::node v) const {
+    std::vector<NetworKit::node> neighborsPlus = getNeighbors(v);
+    neighborsPlus.push_back(v);
+    return neighborsPlus;
+}
+
+std::vector<NetworKit::node> IndependentSet::getNeighbors2(NetworKit::node v) const {    
+    std::vector<NetworKit::node> neighbors = getNeighbors(v);
+    std::vector<NetworKit::node> visitedNodes;    
+    for (auto n : neighbors) {
+        graph->forNeighborsOf(n, [&](NetworKit::node n2) {
+            visitedNodes.push_back(n2);
+        });
+    }
+
+    std::vector<bool> visitedOnGraph(graph->numberOfNodes());
+    visitedOnGraph[v] = true;
+    for (auto x : neighbors) {
+        visitedOnGraph[x] = true;
+    }
+
+    std::vector<NetworKit::node> neighbors2;
+    for (auto x : visitedNodes) {
+        if (!visitedOnGraph[x]) {
+            visitedOnGraph[x] = true;
+            neighbors2.push_back(x);
+        }
+    }
+    return neighbors2;
```

Comment:
Wouldn't it be simpler to prepare (1) set of neighbors of neighbors, (2) set of NeighborsPlus - and then subtract 2 from 1?